src/BizHawk.Client.EmuHawk/CustomControls/ClipboardEventTextBox.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Windows.Forms;
+using BizHawk.Common;
+
+namespace BizHawk.Client.EmuHawk.CustomControls
+{
+	public class ClipboardEventTextBox : TextBox
+	{
+		protected override void WndProc(ref Message m)
+		{
+			const int WM_PASTE = 0x302;
+
+			if (m.Msg == WM_PASTE && !OSTailoredCode.IsUnixHost)
+			{
+				if (OnPasteInternal())
+				{
+					return;
+				}
+			}
+
+			base.WndProc(ref m);
+		}
+
+		protected override bool ProcessCmdKey(ref Message m, Keys keyData)
+		{
+			if (OSTailoredCode.IsUnixHost && keyData is (Keys.Control | Keys.V) or (Keys.Shift | Keys.Insert) && !ReadOnly)
+			{
+				return OnPasteInternal();
+			}
```

Comment:
Alright, second try. Hopefully this works on Linux.

Reply:
Yep. And as expected, right-click > `Paste` doesn't.

Reply:
Yeah I can't think of anything to do about that without finding a completely different approach. I'm afraid this is as good as my PR will get on Linux